### ISSUE TITLE
segfault in parquet tests on Windows

### ISSUE THREAD

**Issue by TomAugspurger** [dask/dask#6093]
https://github.com/dask/dask/pull/6090/checks?check_run_id=584544978

```
dask\dataframe\io\tests\test_parquet.py ........xx..............s....... [ 44%]
......................................................x...........x..... [ 45%]
Windows fatal exception: access violation

Thread 0x0000072c (most recent call first):
  File "C:\Miniconda3\envs\test-environment\lib\multiprocessing\pool.py", line 576 in _handle_results
  File "C:\Miniconda3\envs\test-environment\lib\threading.py", line 870 in run
  File "C:\Miniconda3\envs\test-environment\lib\threading.py", line 932 in _bootstrap_inner
  File "C:\Miniconda3\envs\test-environment\lib\threading.py", line 890 in _bootstrap
```

Will look into this a bit (I'm booting up a windows VM to track down this and the other flaky tests).

**Comment by TomAugspurger** [comment]
I'm able to reproduce the segfault. It's in test_to_parquet_default_writes_nulls.

From the Arrow tracker, https://issues.apache.org/jira/browse/ARROW-8385 sounded somewhat similar but I can't reproduce it, so it's seemingly something else.

**Comment by TomAugspurger** [comment]
Interestingly, the issue isn't parquet, it's in concatenting the tables after their in memory.

```python
import pyarrow as pa
import pandas as pd

t = pa.Table.from_pandas(pd.DataFrame({"A": [1, 2]}))
print("concat")
pa.lib.concat_tables([t])

print('done')
```

https://issues.apache.org/jira/browse/ARROW-8462 has the issue. For now, I'm going to try pinning pyarrow to 0.15.1 on Windows.

**Comment by TomAugspurger** [comment]
This is fixed in pyarrow nightly (>=0.16.1 I think). I think the skip in https://github.com/dask/dask/pull/6094 is what we want long-term (skipping exactly 0.16.0), so I'll close this now.
